app/src/main/java/chat/rocket/android/server/domain/GetChatRoomsInteractor.kt

```diff
@@ -0,0 +1,22 @@
+package chat.rocket.android.server.domain
+
+import chat.rocket.core.model.ChatRoom
+import kotlinx.coroutines.experimental.async
+import javax.inject.Inject
+
+class GetChatRoomsInteractor @Inject constructor(private val repository: ChatRoomsRepository) {
+    fun get(url: String) = repository.get(url)
+
+    suspend fun getByName(url: String, name: String): List<ChatRoom> {
+        val chatRooms = async {
+            val allChatRooms = repository.get(url)
+            if (name.isEmpty()) {
+                return@async allChatRooms
+            }
+            return@async allChatRooms.filter {
+                it.name.contains(name, true)
+            }
```

Comment:
Will this one also filter DMs?

Reply:
Yes, chatrooms includes all types of "rooms"

Reply:
👍